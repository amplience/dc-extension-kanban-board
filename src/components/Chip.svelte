<script lang="ts">
  import Truncate from './Truncate.svelte';
  export let backgroundColor: string = '';
  export let color: string = '';
  export let label: string = '';
</script>

<style lang="scss">
  .chip {
    min-width: 120px;
    max-width: 227px;
    margin: 0 2px 0 10px;
    font-size: 13px;
    box-sizing: border-box;
    border-radius: 9px;
    color: white;
    padding: 0.15em 0.5em;
    display: inline-block;
    margin: 0.5em;
    text-align: center;
    height: 24px;
    line-height: 24px;

    &.light {
      color: #f2f2f2;
    }

    &.dark {
      color: #333333;
    }
  }
</style>

<div class="chip {color}" style="background-color: {backgroundColor}">
  <Truncate text={label} />
</div>
