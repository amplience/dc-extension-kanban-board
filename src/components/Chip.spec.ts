import { render } from '@testing-library/svelte';
import Chip from './Chip.svelte';

describe('chip component', () => {
  it('should render a chip component', () => {
    const { container } = render(Chip, {
      label: 'LABEL',
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});
