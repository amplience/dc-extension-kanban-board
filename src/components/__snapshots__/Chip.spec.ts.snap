// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chip component should render a chip component 1`] = `
<div>
  <div
    class="chip  svelte-8agz7l"
  >
    <div
      class="truncate svelte-4ya5ts"
      title="LABEL"
    >
      LABEL
    </div>
  </div>
</div>
`;
